### PR TITLE
refactor/move-kernelfunc

### PR DESCRIPTION
**Pull request by djacu**
Extract `kernelFunc` to modules/poetry.nix.

### CODE REVIEW

**Comment by GTrunSec** [comment]
Notice: https://github.com/tweag/jupyenv/pull/450#issuecomment-1455026847, which means I can not import such as https://github.com/tweag/jupyenv/blob/main/lib/overrides.nix from the evalModule.(I don't why).

my solution: https://github.com/tweag/jupyenv/pull/450/files#diff-8bf585666371b85c4eb513ad2163c69e20eaf72b2faa18abc1a0884c9827d70fR136

Also, I think it is necessary to provide an additional option for poetryEnv, not only the problem of overalys, the most important thing is that pythonEnv can be passed from the outside. (we can simply add a group deps of jupyenv for other poetyEnv ,  then add it)

File: modules/poetry.nix [hunk 1]
```diff
@@ -19,9 +21,78 @@
...
+      poetrylock,
+      overrides,
+      python,
+      editablePackageSources,
+      extraPackages,
+      preferWheels,
+      groups,
+      ignoreCollisions,
+    }: let
+      env =
```
Comment (GTrunSec) [inline modules/poetry.nix:52]: we should move the env to the moduleType totally, and make sure the users are able to rescue the poetyEnv and pure python support. like that https://github.com/GTrunSec/jupyterWith/blob/dev/modules/types/poetry.nix#L136

File: modules/poetry.nix [hunk 2]
```diff
@@ -19,9 +21,78 @@
   }: let
     args = {inherit self system lib config name kernelName requiredRuntimePackages;};
     kernelModule = import ./kernel.nix args;
+
+    kernelFunc = {
```
Comment (GTrunSec) [inline modules/poetry.nix:25]: we only keep some necessary args in the `KernelFunc`. (simplify it again) 
 https://github.com/GTrunSec/jupyterWith/blob/dev/kernels/available/python/default.nix#L7

Comment (GTrunSec) [reply modules/poetry.nix:25]: fixes #485

File: modules/poetry.nix (hunk 1 above)
Comment (GTrunSec) [reply modules/poetry.nix:52]: fixes https://github.com/tweag/jupyenv/issues/485

**Review by balsoft — APPROVED** [review]

**Pull request merged**